rewrite-maven/src/main/java/org/openrewrite/maven/search/FindProperties.java

```diff
@@ -55,18 +55,18 @@ public String getDescription() {
 
     @Override
     public TreeVisitor<?, ExecutionContext> getVisitor() {
-        Pattern propertyMatcher = Pattern.compile(propertyPattern.replace(".", "\\.")
-                .replace("*", ".*"));
+        Pattern propertyMatcher = Pattern.compile(propertyPattern);
+        Pattern propertyUsageMatcher = Pattern.compile(".*\\$\\{" + propertyMatcher.pattern() + "}.*");
         return new MavenVisitor<ExecutionContext>() {
             @Override
             public Xml visitTag(Xml.Tag tag, ExecutionContext context) {
                 Xml.Tag t = (Xml.Tag) super.visitTag(tag, context);
-                if (isPropertyTag() && propertyMatcher.matcher(tag.getName()).matches()) {
+                if (isPropertyTag() && propertyMatcher.matcher(t.getName()).matches()) {
                     t = SearchResult.found(t);
                 }
 
-                Optional<String> value = tag.getValue();
-                if (t.getContent() != null && value.isPresent() && value.get().contains("${")) {
+                Optional<String> value = t.getValue();
+                if (value.isPresent() && propertyUsageMatcher.matcher(value.get()).matches()) {
```

Comment:
No need to check for `t.getContent != null` since it's already checked in `t.getValue()` returning an `Optional.empty()`